docs/examples/day-night-masking.py

```diff
@@ -0,0 +1,125 @@
+# if a time series has daylight savings time or time shifts. Here, we use
+# :py:func:`pvanalytics.features.daytime.power_or_irradiance` to mask day/night
+# periods, as well as estimate sunrise and sunset times in the data set. This
+# function is particularly useful for cases where the time zone of a data
+# stream is unknown or incorrect.
+
+import pvanalytics
+from pvanalytics.features.daytime import power_or_irradiance
+import matplotlib.pyplot as plt
+import pandas as pd
+import pathlib
+import pvlib
+import numpy as np
+
+# %%
+# First, read in the 1-minute sampled AC power time series data, taken
+# from the SERF East installation on the NREL campus.
+# This sample is provided from the NREL PVDAQ database, and contains
+# a column representing an AC power data stream.
+
+pvanalytics_dir = pathlib.Path(pvanalytics.__file__).parent
+ac_power_file = pvanalytics_dir / 'data' / 'serf_east_1min_ac_power.csv'
+data = pd.read_csv(ac_power_file, index_col=0, parse_dates=True)
+data = data.sort_index()
+# This is the known frequency of the time series. You may need to infer
+# the frequency or set the frequency with your AC power time series.
+freq = "1T"
+# These are the latitude-longitude coordinates associated with the
+# SERF East system.
+latitude = 39.742
+longitude = -105.173
+# Plot the time series.
+data['ac_power__752'].plot()
+plt.xlabel("Date")
+plt.ylabel("AC Power (kW)")
+plt.tight_layout()
+plt.show()
+
+# %%
+# It is critical to set all negative values in the AC power time series to 0
+# for :py:func:`pvanalytics.features.daytime.power_or_irradiance` to work
+# properly. Negative erroneous data may affect daytime mask assignments.
+data.loc[data['ac_power__752'] < 0, 'ac_power__752'] = 0
+
+# %%
+# Now, use :py:func:`pvanalytics.features.daytime.power_or_irradiance`
+# to identify day periods in the time series.
+predicted_day_night_mask = power_or_irradiance(series=data['ac_power__752'],
+                                               freq=freq)
+
+# %%
+# Function :py:func:`pvlib.solarposition.sun_rise_set_transit_spa` is
+# used to get ground-truth sunrise and sunset times for each day at the site
+# location, and a SPA-daytime mask is calculated based on these times. Data
+# associated with SPA daytime periods is labeled as True, and data associated
+# with SPA nighttime periods is labeled as False.
+# SPA sunrise and sunset times are used here as a point of comparison to the
+# sensor-based AC power data; SPA-based sunrise and sunset values are not
+# needed to run :py:func:`pvanalytics.features.daytime.power_or_irradiance`.
+
+sunrise_sunset_df = pvlib.solarposition.sun_rise_set_transit_spa(data.index,
+                                                                 latitude,
+                                                                 longitude,
+                                                                 how='numpy',
+                                                                 delta_t=67.0)
```

Comment:
I'd delete the optional kwargs, no reason to suggest to a user of this example that values other than the default are needed.